service/awsconfig/v10/resource/s3bucket/current.go

```diff
@@ -15,31 +17,45 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for the S3 bucket")
+	r.logger.LogCtx(ctx, "level", "debug", "message", "looking for the S3 buckets")
 
 	accountID, err := r.awsService.GetAccountID()
 	if err != nil {
 		return nil, microerror.Mask(err)
 	}
 
-	bucketName := key.BucketName(customObject, accountID)
-	headInput := &s3.HeadBucketInput{
-		Bucket: aws.String(bucketName),
-	}
-	_, err = r.clients.S3.HeadBucket(headInput)
-	if IsBucketNotFound(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "did not find the S3 bucket")
-		return BucketState{}, nil
-	}
-	if err != nil {
-		return BucketState{}, microerror.Mask(err)
+	bucketState := []BucketState{
+		BucketState{
+			Name:           key.TargetLogBucketName(customObject),
+			IsDeliveryLog:  true,
+			LoggingEnabled: true,
+		},
+		BucketState{
+			Name:           key.BucketName(customObject, accountID),
+			IsDeliveryLog:  false,
+			LoggingEnabled: true,
+		},
 	}
 
-	bucketState := BucketState{
-		Name: bucketName,
+	currentBucketState := []BucketState{}
+
+	for _, inputBucket := range bucketState {
+		headInput := &s3.HeadBucketInput{
+			Bucket: aws.String(inputBucket.Name),
+		}
+		_, err = r.clients.S3.HeadBucket(headInput)
+		if IsBucketNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find the S3 bucket %q", inputBucket.Name))
+		}
+		if err != nil {
+			return []BucketState{}, nil
+		}
+		if err == nil {
+			currentBucketState = append(currentBucketState, inputBucket)
+		}
```

Comment:
Well if the error is not found we leave it empty because it means the current state has no buckets, right?

In case a different error is produced then I agree